content/blog/2024/2024-06-05-aws-s3-presigned-urls.md

```diff
@@ -0,0 +1,468 @@
+---
+title: "Offloading File Transfers with Amazon S3 Presigned URLs in Spring Boot"
+categories: [ "AWS", "Spring Boot", "Java" ]
+date: 2024-06-07 00:00:00 +0530
+modified: 2024-06-07 00:00:00 +0530
+authors: [ "hardik" ]
+description: "In this article, we demonstrate how to use AWS S3 Presigned URLs in a Spring Boot application to offload file transfers, reduce server load and improve performance. We cover required dependencies, configuration, IAM policy, generation of Presigned URLs and integration testing with LocalStack and Testcontainers."
+image: images/stock/0139-stamped-envelope-1200x628-branded.jpg
+url: "offloading-file-transfers-with-amazon-s3-presigned-urls-in-spring-boot"
+---
+
+When building web applications that involve file uploads or downloads, a common approach is to have the files pass through an application server. However, this can lead to **increased load on the server, consuming valuable computing resources, and potentially impacting performance**. A more efficient solution is to **offload file transfers to the client using Presigned URLs**.
+
+In the context of this article, the client refers to the single page application (SPA), desktop, or mobile application that the user is interacting with, and not the end user themselves.
```

Comment:
A SPA, desktop app, or mobile application is essentially the end-user. A BFF, for example, would be a different story.

Reply:
got it, let's use the original version of the paragraph then.